channel.go

```diff
@@ -29,27 +35,34 @@ type channel interface {
 }
 
 func newChannel() (channel, error) {
```

Comment:
It'd be nice to add a verbose comment on top of this function. We need to explicitly say that the agent will try by default to connect to a serial port, and that a failure doing so will not result in an agent failure, but instead it will fallback to the mechanism of waiting for a VSock socket to be created, and this will be gated by a timeout.